app/controllers/metrics_controller.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+# MetricsController is responsible for handling requests related to metrics,
+# including exposing SendGrid-related metrics in a format compatible with Prometheus.
+class MetricsController < ApplicationController
+  def index # rubocop:disable Metrics/MethodLength
+    result = SendgridService.new.check_credits
```

Comment:
**issue**: The Prometheus instance is scraping the `/metrics` endpoint every 30s
(I verified here: https://prometheus.simple.org/config)

As such, Prometheus scrapes are always happening at very fast intervals—we don't need to shoot off API calls to collect balance for every scrape, and we may risk hitting rate limits as well.

**suggestion**: Instead, we can setup a scheduled job that will observe the gauge every hour (see how we check SMS balances in `scheduler.rb`)

Reply:
@tfidfwastaken Arnaud asked me to do this and not to use schdeuled job for service.